STEP_Project/tests/OUListContentTest.js

```diff
@@ -2,160 +2,172 @@ let tabl;
 let orgUnits;
 let numRows;
 
-describe('Testing the initButtons function', ()=>{
+describe('Testing the initButtons and addButtons function', ()=>
+{
     beforeEach(function()
     {
-        tabl = document.createElement('table');
         numRows = 3;
+        tabl = document.createElement('table');
+        tabl.setAttribute("role", "grid");
         for (let i = 0; i < numRows; i++)
         {
             tabl.insertRow();
         }
         orgUnits = tabl.rows;
-
-    })
-
-    afterEach(function()
-    {
-        for (let i = 0; i < numRows; i++)
+        for(let i = 0; i < numRows; i++)
         {
-            tabl.deleteRow(0);
+            orgUnits[i].setAttribute("data-row-id", i);
         }
+        document.body.appendChild(tabl);
     })
 
-    it('Should not add a button to the first row', ()=>{
-        initButtons(tabl);
-        let row = orgUnits[0];
-        let temp = row.getElementsByClassName('bClass');
-        expect(temp.length).toEqual(0);
-    })
-
-    it('Should add a button to every row after that', ()=>{
-        initButtons(tabl);
-        let row = orgUnits[1];
-        let temp = row.getElementsByClassName('bClass');
-        expect(temp.length).toEqual(1);
-        row = orgUnits[2];
-        temp = row.getElementsByClassName('bClass');
-        expect(temp.length).toEqual(1);
-    })
-})
-
-describe('Testing the addButtons function', ()=>{
-    beforeEach(function()
+    afterEach(function()
     {
-        numRows = 3;
-        tabl = document.createElement('table');
-        for (let i = 0; i < numRows; i++)
-        {
-            tabl.insertRow();
-        }
-        orgUnits = tabl.rows;
-        initButtons(tabl);
+        tabl.remove();
     })
 
-    afterEach(function()
+    it('Should add "+" buttons to all rows except first, should store OU on clicking "+" button and change button to "-", should remove OU from storage on clicking "-"', ()=>
```

Comment:
Please break it into multiple tests testing individual functionality. I think it can this be broken down into at least 3 tests, example: addButtons (both +/-), addRemoveButtonClick with just + button clicked and addRemoveButtonClick with just - button clicked. You might have to create separate DOM before each test for this to work.

Reply:
Split into 5 tests, updating description with the most recent screenshot